src/reconciler/exec/external.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+use vstd::{prelude::*, view::*};
+
+verus! {
+
+// A trait for the external library of a reconciler.
+// Its core is a process method, and the developer should wrap all possible operations they may need in the function.
+// The InputType is the ? of Request<?> of the reconciler, i.e., it completes the request type of a reconciler.
+// Similarly, the OutputType composes the Response<?> type of a reconciler.
+// Note that we can encapsulate all the required libraries here, so each reconciler only has one ExternalLibrary type.
+pub trait ExternalLibrary<InputType: View, OutputType: View> {
+    fn process(input: InputType) -> Option<OutputType>;
```

Comment:
Sometimes the third party APIs are async functions. Since process is not async, we probably should check whether Verus supports async block (in external functions) so that we can call async functions in process

Reply:
What does that mean? I think we have already used some asyn blocks.

Reply:
You mean the one in `run_controller`
```
    let reconcile = |cr: Arc<K>, ctx: Arc<Data>| async move {
        return reconcile_with::<K, ResourceWrapperType, ReconcilerType, ReconcileStateType>(
            &ReconcilerType::default(), cr, ctx
        ).await;
    };
```
?

Reply:
It is an async closure but not an async block, tho they are very similar